provider/kubernetes/kubernetes.go

```diff
@@ -294,6 +291,25 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 	return &templateObjects, nil
 }
 
+func setPath(pa v1beta1.HTTPIngressPath, i *v1beta1.Ingress, templateObjects types.Configuration, r v1beta1.IngressRule) {
```

Comment:
I twiddled the implementation from `setPath` to `getRuleForPath`, and, assuming a rule is found, apply it.  That should address the two issues you highlighted.